### ISSUE TITLE
Line Graphs (I2g)

### ISSUE THREAD

**Issue by dlee138** [Statistical-Connectomics-Sp15/intro#133]
So according to the paper a line graphs are essentially graphs where vertices and edges have been reversed. Was this switch necessary to access graph error in the i2g paper?  Was their explanation basically saying that synapses are denser, and therefore less prone to fragmentation, making them a better choice for nodes?  Could the conventional graph form have worked as well and how do you think it would perform?

**Comment by edunnwe1** [comment]
I believe the key feature is the correspondence between the true and the estimated graphs. They specify that the graphs should have a direct correspondence. Since the dendritic spines and thus synaptic connections are small, we can imagine that their presence is effectively binary: there either is a spine or there is not. This is helpful for direct correspondence because it makes it easy to identify what synapses are shared by both graphs, and what synapses are missing by one or the other graph. So, you can clearly define spurious entries via false positive or false negative entries in their method (p3)

To use this direct correspondence with the conventional graph, you would again begin by assigning a direct correspondence between the neurons picked up by each of the graph. But, say the true graph picked up neuron A and the estimated graph picked up A' and A'' two fragments of neuron A. This can't be turned into a 1-1 correspondence in an obvious way: how do you define false positive or false negative here? Thus I think that the dual graph (line graph) is more elegant and straightforward.

**Comment by willgray** [comment]
Hey guys!  I missed this question.  The correspondence is indeed a big thing - more on the synapses than the neighboring spines.  Graph matching is a tough problem, and by using the line graph we sidestep this problem and can directly compare the matrices (we line up synapse correpondence through overlap - more on that if you want it).  

I'm not sure how to do error properly in the conventional graph.  Split errors are a problem - merge errors make things even worse.  One other note is that although edges become nodes in the line graph, the edges in the line graphs are paths (not necessarily 1-1 with neurons).

Let me know if you disagree or have more questions!!

**Comment by edunnwe1** [comment]
Cool! Can you explain your last note that edges in the line graphs are paths (not 1-1 with neurons)?

**Comment by willgray** [comment]
Yup!  So line graphs show connections/paths between synapses.  So each path
is a part of a neuron, but many possible paths can be drawn between two
points on a neuron...so think of an edge in a line graph as a neuron
fragment.

Is that clear or worse?

Best,
Will

**Comment by gkiar** [comment]
Revisiting Figure 4 in the paper after the explanation may help :)
http://arxiv.org/pdf/1411.6880v1.pdf

--
Greg Kiar <gkiar07@gmail.com>

**Comment by edunnwe1** [comment]
Yea the figure was actually part of what made it unintuitive since there you do have the same number of lines as you have neurons. But is the idea that in the original graph, the nodes are neuron fragments rather than neurons and now those nodes are edges? So there is still a 1:1 relationship between edges of the line graph and nodes of the original graph its just that neither of these are neurons? Or am I getting it mixed up? thanks :)

**Comment by willgray** [comment]
Not a 1:1 relationship (unfortunately?)

So edges in the line graph represent paths.  A single neuron fragment can
be used to complete multiple two way paths (for example, three synapses on
the same branch of a dendrite).  Or no paths at all, if there are no
synapses attached.

Does that help?  If not, I have lots to say...


Best,
Will

**Comment by edunnwe1** [comment]
Yes, that does help, thank you!!
